src/IECoreImage/ImageReader.cpp

```diff
@@ -343,34 +326,44 @@ class ImageReader::Implementation
 		// Exception is thrown rather than false being returned.
 		bool open( bool throwOnFailure = false )
 		{
-			if( m_inputFile && m_reader->fileName() == m_inputFileName )
+			if( m_cache && m_reader->fileName() == m_inputFileName )
 			{
 				// we already opened the right file successfully
 				return true;
 			}
 
 			m_inputFileName = "";
-			m_inputFile.reset( ImageInput::open( m_reader->fileName() ) );
-			if( !m_inputFile )
+			m_cache.reset( ImageCache::create( /* shared */ false ) );
+
+			if( ImageInput *input = ImageInput::create( m_reader->fileName() ) )
```

Comment:
Is this necessary? It seems wrong that we're using both the ImageCache _and_ direct access to ImageInput. In Gaffer we appear to [get the file format from the image cache](https://github.com/GafferHQ/gaffer/blob/master/src/GafferImage/OpenImageIOReader.cpp#L615) itself.

Reply:
Oh, nice. I'll try that out.

Reply:
Addressed in 73be01e